### PR TITLE
Pytests reader

### PR DESCRIPTION
**Pull request by bryantChhun**
conftests and pytests for the reader class plus its subclasses.

also contains a bug fix in the reader where metadata is string-parsed to create coordinate-filename maps

### CODE REVIEW

File: tests/reader/test_multipagetiff.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    mmr = MicromanagerOmeTiffReader(rand_folder, extract_data=True)
+    for i in range(mmr.get_num_positions()):
+        z = mmr.get_zarr(i)
+        assert(z.shape == mmr.shape)
+
+
+def test_get_array_mm2gamma(setup_mm2gamma_ome_tiffs):
+    _, _, rand_folder = setup_mm2gamma_ome_tiffs
+    mmr = MicromanagerOmeTiffReader(rand_folder, extract_data=True)
+    for i in range(mmr.get_num_positions()):
```
Comment (camFoltz) [inline tests/reader/test_multipagetiff.py:51]: For the get_array and get_zarr tests, do we want to assert that one is a zarr object and one is a numpy array?  This test could pass if the get_array returned a zarr array or vice versa

Comment (bryantChhun) [reply tests/reader/test_multipagetiff.py:51]: good idea, will do

Comment (bryantChhun) [reply tests/reader/test_multipagetiff.py:53]: done

**Review by camFoltz — APPROVED** [review]
Looks good, all the tests passed on my end

**Pull request merged**